src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -31,32 +28,39 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             try
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
-                ConfigurationClient client;
+                IConfigurationClientProvider clientProvider;
 
-                if (options.Client != null)
+                if (options.ClientProvider != null)
                 {
-                    client = options.Client;
+                    clientProvider = options.ClientProvider;
                 }
                 else if (!string.IsNullOrWhiteSpace(options.ConnectionString))
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.ConnectionString, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.ConnectionString, options.ClientOptions);
                 }
-                else if (options.Endpoint != null && options.Credential != null)
+                else if (options.Endpoints != null && options.Credential != null)
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.Endpoint, options.Credential, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.Endpoints, options.Credential, options.ClientOptions);
                 }
                 else
                 {
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                provider = new AzureAppConfigurationProvider(clientProvider, options, _optional);
             }
-            catch (InvalidOperationException e)
+            catch (InvalidOperationException ex)
             {
                 if (!_optional)
                 {
-                    throw new ArgumentException(e.Message, e);
+                    throw new ArgumentException(ex.Message, ex);
+                }
+            }
+            catch (FormatException fe)
```

Comment:
It will be helpful to have some comment about when we will get InvalidOperationException vs. FormatException. The handling code is the same. Can we combine them?

Reply:
Done!